node/src/syncer.rs

```diff
@@ -1,16 +1,60 @@
 use std::sync::Arc;
 
+use async_trait::async_trait;
 use tokio::sync::RwLock;
 
-use crate::store::Store;
+use crate::{p2p::P2pService, store::Store, Service};
+
+type Result<T, E = SyncerError> = std::result::Result<T, E>;
+
+#[derive(Debug, thiserror::Error)]
+pub enum SyncerError {}
 
 #[allow(unused)]
-pub struct Syncer {
+#[derive(Debug)]
+pub struct Syncer<P2pSrv: P2pService> {
+    p2p: Arc<P2pSrv>,
     store: Arc<RwLock<Store>>,
 }
 
-impl Syncer {
-    pub fn new(store: Arc<RwLock<Store>>) -> Self {
-        Syncer { store }
+pub struct SyncerArgs<P2pSrv: P2pService> {
+    pub p2p: Arc<P2pSrv>,
+    pub store: Arc<RwLock<Store>>,
+}
+
+#[derive(Debug)]
+pub enum SyncerCmd {}
```

Comment:
Same as `P2pCmd`, this should be private

Reply:
hidden